### PR TITLE
Clean up client<>server synchronization in stream test

### PR DESCRIPTION
**Pull request by jack-kearney**
## Summary & Motivation (Problem vs. Solution)
The `stream_implements_read_write_traits` was failing locally. This makes the server<>client interaction less flakey

## How I Tested These Changes
Locally

## Pre merge check list

- [ ] Update CHANGELOG.MD

### CODE REVIEW

File: src/qos_core/src/io/stream.rs [hunk 1]
```diff
@@ -372,6 +372,7 @@ mod test {
 		os::{fd::AsRawFd, unix::net::UnixListener},
 		path::Path,
 		str::from_utf8,
+        sync::{Arc, Barrier},
```
Comment (r-n-o) [inline src/qos_core/src/io/stream.rs]: Indentation seems off

File: src/qos_core/src/io/stream.rs [hunk 2]
```diff
@@ -446,9 +455,10 @@ mod test {
 		// request
 		let mut server =
 			HarakiriPongServer::new(socket_server_path.to_string());
-		thread::spawn(move || {
-			server.start();
-		});
+        let barrier = server.start();
+		// thread::spawn(move || {
+		// 	server.start();
+		// });
```
Comment (r-n-o) [inline src/qos_core/src/io/stream.rs]: Did you mean to comment this out? I think we can remove this

File: src/qos_core/src/io/stream.rs (hunk 1 above)
Comment (jack-kearney) [reply src/qos_core/src/io/stream.rs]: Thx fixed

File: src/qos_core/src/io/stream.rs (hunk 2 above)
Comment (jack-kearney) [reply src/qos_core/src/io/stream.rs]: Removed

File: src/qos_core/src/io/stream.rs [hunk 3]
```diff
@@ -391,9 +392,14 @@ mod test {
...
 		}
-		pub fn start(&mut self) {
+		pub fn start(&mut self) -> Arc<Barrier> {
 			let listener = UnixListener::bind(&self.path).unwrap();
 			let path = self.path.clone();
+
+            // Create a barrier to synchronize between server and client
+            let barrier = Arc::new(Barrier::new(2));
+            let server_barrier = Arc::clone(&barrier);
+
```
Comment (r-n-o) [inline src/qos_core/src/io/stream.rs:402]: Same here, indentation is off (tabs vs spaces). 

This tabs v space issue is there for all new lines you inserted so it's probably an editor configuration issue.

**Pull request closed without merging**